das-query-engine/validators/actions.py

```diff
@@ -35,30 +35,41 @@ class GetLinksValidator(PayloadValidator):
     target_types = datatypes.Array(required=False)
     link_targets = datatypes.Array(required=False)
     kwargs = datatypes.Function(
-        lambda value, *args, **kwargs: isinstance(value, dict) if value is not None else True,
+        lambda value, *args, **kwargs: (
+            isinstance(value, dict) if value is not None else True
+        ),
         required=False,
     )
 
+
 class GetIncomingLinksValidator(PayloadValidator):
     strict = True
 
     atom_handle = datatypes.String(required=True)
     kwargs = datatypes.Function(
-        lambda value, *args, **kwargs: isinstance(value, dict) if value is not None else True,
+        lambda value, *args, **kwargs: (
+            isinstance(value, dict) if value is not None else True
+        ),
         required=False,
     )
 
 
-class QueryValidator(PayloadValidator):
+class QueryInputValidator(PayloadValidator):
+    class QueryValidator(PayloadValidator):
```

Comment:
This `QueryValidador` is wrong. You're only considering the `link type` format. But the user can pass a `node type`. Some like that:

```
q = {'atom_type': 'node', 'type': 'Symbol', 'name': 'Blah'}

das.query(query=q)
```

Reply:
@marcocapozzoli ok, I've fix it